frontend/src/metabase/query_builder/components/CompareAggregations/components/ColumnPicker/ColumnPicker.tsx

```diff
@@ -15,39 +16,26 @@ interface ItemType {
 }
 
 interface Props {
+  comparisonType: ComparisonType;
   value: ColumnType[];
   onChange: (value: ColumnType[]) => void;
 }
 
-const COLUMN_OPTIONS: ItemType[] = [
-  {
-    example: "1826, 3004",
-    label: t`Previous value`,
-    value: "offset",
-  },
-  {
-    example: "+2.3%, -0.1%",
-    label: t`Percentage difference`,
-    value: "percent-diff-offset",
-  },
-  {
-    example: "+42, -3",
-    label: t`Value difference`,
-    value: "diff-offset",
-  },
-];
-
-export const ColumnPicker = ({ value, onChange }: Props) => {
+export const ColumnPicker = ({ value, onChange, comparisonType }: Props) => {
   const handleChange = useCallback(
     (values: string[]) => {
       onChange(values as ColumnType[]);
     },
     [onChange],
   );
 
+  useDeepCompareEffect(() => {
```

Comment:
effects are usually hard to follow. Can we do this conversion of column types in event handlers instead?